src/pages/index.js

```diff
@@ -6,87 +6,230 @@ import {
-const cardSectionClass = new Section(
-  {
-    items: cardData,
-    renderer: renderCard,
-  },
-  selectors.cardSection
-);
+
 const editFormValidator = new FormValidator(validationSettings, editFormEl);
 const addFormValidator = new FormValidator(validationSettings, addFormEl);
+const pictureFormValidator = new FormValidator(
+  validationSettings,
+  pictureFormEl
+);
 
 // INTITIALIZE //
-cardSectionClass.renderItems(cardData);
 cardPreviewClass.setEventListeners();
 profileFormClass.setEventListeners();
 cardAddFormClass.setEventListeners();
-
+pictureFormClass.setEventListeners();
+confirmFormClass.setEventListeners();
 editFormValidator.enableValidation();
 addFormValidator.enableValidation();
+pictureFormValidator.enableValidation();
 
 // FUNCTIONS //
 function handleImageClick(card) {
   cardPreviewClass.open({ name: card._name, link: card._link });
 }
 
-function handleCardAddSubmit(data) {
-  renderCard({ name: data.title, link: data.link });
-  cardAddFormClass.close();
-  cardAddForm.reset();
+function handleCardAddSubmit(inputData) {
+  cardAddFormClass.isLoading(true);
+  api
+    .addCard(inputData)
+    .then((data) => {
+      const { _id, name, link, isLiked } = data;
+      const newCardData = { _id, name, link, isLiked };
+      cardSection.addItem(createCard(newCardData));
+      addFormValidator.disableBtn();
+      cardAddForm.reset();
+      cardAddFormClass.close();
+    })
+    .catch((err) => {
+      console.error(`Error: failed to add card. ${err}`);
+    })
+    .finally(() => {
+      cardAddFormClass.isLoading(false);
+    });
 }
 
 function handleProfileFormSubmit(data) {
-  profileInfoClass.setUserInfo(data);
-  profileFormClass.close();
+  console.log(data);
+  profileFormClass.isLoading(true);
+  api
+    .setUserInfo(data)
+    .then(() => {
+      profileInfoClass.setUserInfo({ name: data.name, about: data.about });
+      editFormValidator.disableBtn();
+      profileFormClass.close();
+    })
+    .catch((err) => {
+      console.error(`Error handling profile form submit. ${err}`);
+    })
+    .finally(() => {
+      profileFormClass.isLoading(false);
+    });
 }
 
-function renderCard(card) {
-  const cardElementClass = new Card({
-    name: card.name,
-    link: card.link,
-    cardSelector: "#card-template",
-    handleImageClick: handleImageClick,
+function handlePictureSubmit(avatarUrl) {
+  pictureFormClass.isLoading(true);
+  api
+    .changeAvatar(avatarUrl)
+    .then(() => {
+      profileInfoClass.setUserInfo({ avatar: avatarUrl.picture });
+      pictureFormValidator.disableBtn();
+      pictureFormClass.close();
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      pictureFormClass.isLoading(false);
+    });
+}
+
+function handleLikeBtn(card) {
+  api
+    .setCardLiked(card.id, card.getLikedStatus())
+    .then((updatedCard) => {
+      card.handleIsLiked(updatedCard.isLiked);
+    })
+    .catch((err) => {
+      console.error(err);
```

Comment:
Good job that you `catch` possible errors at the end of server requests. You can simplify it with `.catch(console.error);` because `console.error` is meant to log errors